### PR TITLE
replaced two-step processing part

### LINKED ISSUE #74: two-step processing method to two-level processing method?
In Japanese text, `two-step processing method` is written as two level. Also, `step` usually refers each detailed procedure within one process like API implementation. 
This could be confusing and we might be better to consider of rewording `two-step processing method` with `level` or something.

### PR DESCRIPTION
**Pull request by himorin**
closes #74, NOTE: conflicts to #75

use `ruby box` and `two level`, with new text from Bin-sensei.
still wip.

original Japanese text: https://lists.w3.org/Archives/Public/public-i18n-japanese/2022JanMar/0037.html
but could be updated later.

***
<a href="https://pr-preview.s3.amazonaws.com/himorin/simple-ruby/pull/84.html" title="Last updated on Apr 20, 2023, 6:53 AM UTC (118462c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/simple-ruby/84/bce1a03...himorin:118462c.html" title="Last updated on Apr 20, 2023, 6:53 AM UTC (118462c)">Diff</a>

### CODE REVIEW

**Review by murata2makoto — COMMENTED** [review]
I am not sure if I can improve Shimono-san's English.  But let me try

The first level determines the relative positions of ruby base and ruby annotations and creates ruby boxes.    Meanwhile, the second level determines the position of ruby boxes in lines.  In other words, the relative positions of ruby base and ruby annotation as decided in the first level are not modified in light of any characters or line edges preceding or following ruby boxes.

**Comment by murata2makoto** [comment]
As of now, " Two levels are processed as following, with using ruby column for paired ruby annotation...." is hard to understand.  Do we really need this?

**Comment by himorin** [comment]
> As of now, " Two levels are processed as following, with using ruby column for paired ruby annotation...." is hard to understand. Do we really need this?

Yeah... really wondering how to locally define two words here..

**Comment by himorin** [comment]
drafting back and forth for a while,,, how about something like below?

- two-level processing specific terms are not clearly defined
- align with provided text

```
<li id="l20200529009">Processing is done in two levels.
    The first level only determines relative positions of ruby annotation and its base. 
    Meanwhile, the second level determines inline positions of <a>ruby boxes</a>. 
    In other words, relative positions of the ruby annotation and its base text as decided in the first level 
    are not modified by the second level such as in light of any characters or line edge preceding and following the <a>ruby box</a>.
    Two levels are processed as following:
    <ol>
    <li id="l20220318001">The first level takes pairs of ruby annotation and its base text 
        and parameters required for procesing such as available space in inline direction in the current line, 
        but does not take any preceding or following characters. 
        The first level outputs one or more untransparent <a>ruby boxes</a>, which have relative position of ruby annotation to its base.</li>
    <li id="l20220318002">The second level takes a size of untransparent <a>ruby box</a> and its preceding and following text, 
        and outputs styled text.</li>
    </ol>
</li>
```

File: index.html [hunk 1]
```diff
@@ -210,18 +210,17 @@ <h3 id="matters-considered-by-the-placement-rules">Considerations for  simple pl
...
 <li id="l20200529009">Processing is done in two levels.
-    In the first level, processing of layout only considers the relative position of the ruby annotation and its base text. 
-    In the second level, layout processing decides the position of the <a>ruby box</a> in the line. 
+    The first level only determines relative positions of ruby annotation and its base. 
+    Meanwhile, the second level determines inline positions of <a>ruby boxes</a>. 
     In other words, relative positions of the ruby annotation and its base text as decided in the first level 
     are not modified by the second level such as in light of any characters or line edge preceding and following the <a>ruby box</a>.
-    Two levels are processed as following, with using <dfn>ruby column</dfn> for paired ruby annotation and its base text, 
-    and untransparent <a>ruby box</a> for <a>ruby box</a> with their relative position:
+    Two levels are processed as following:
```
Comment (r12a) [inline index.html:219]: as following -> as follows

File: index.html [hunk 2]
```diff
@@ -210,18 +210,17 @@ <h3 id="matters-considered-by-the-placement-rules">Considerations for  simple pl
...
     <ol>
-    <li id="l20220318001">The first level takes a <a>ruby column</a> with required parameters for 
-        processing placement for inline direction inluding available space within the current line 
-        but without any preceding or following characters, 
-        and outputs one or more untransparent <a>ruby box</a>.</li>
-    <li id="l20220318002">The second level takes a size and preceding and following text of <a>ruby boxes</a>, 
+    <li id="l20220318001">The first level takes pairs of ruby annotation and its base text
```
Comment (r12a) [inline index.html]: Suggested edits:
The first level takes pairs of ruby base and its annotation, including parameters needed for processing, such as available space in the inline direction of the current line. The first level outputs one or more [ruby boxes], that indicate the position of the ruby annotation with respect to its base.
The second level looks at the size of untransparent [ruby box] and its surrounding text, and outputs styled text.

Comment (r12a) [reply index.html]: I'm not clear what 'untransparent' means (i haven't looked around the document for an explanation).

**Review by r12a — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (himorin) [reply index.html:219]:
```suggestion
    Two levels are processed as follows:
```

File: index.html (hunk 2 above)
Comment (himorin) [reply index.html]: Thank you for comment on this. For 'untransparent', there is still discussion on terminology and explanation even in Japanese, and we are seeking better (and clearer?) text for this part...

**Comment by himorin** [comment]
> drafting back and forth for a while,,, how about something like below?

After several offline conversations and discussions,,, how about as following: @murata2makoto @kidayasuo ?

```
<li id="l20200529009">Processing is done in two levels.
     The first level only determines relative positions of ruby annotation and its base.
     Meanwhile, the second level determines inline positions of <a>ruby boxes</a>.
     In other words, relative positions of the ruby annotation and its base text as decided in the first level
     are not referenced nor modified by the second level such as in light of any characters or line edge preceding and following the <a>ruby box</a>.
     Two levels are processed as following:
     <ol>
     <li id="l20220318001">The first level takes pairs of ruby base and its annotation
         including parameters needed for processing, such as available space in the inline direction of the current line,
         but excluding any preceding or following characters to the ruby base.
         The first level outputs one or more <a>ruby boxes</a>, that indicate the outside size of <a>ruby boxes</a>
         and the position of the ruby annotation with respect to its base.</li>
     <li id="l20220318002">The second level looks at the outside size of <a>ruby box</a> and its surrounding text,
         and outputs styled text.</li>
     </ol>
</li>
```

File: index.html (hunk 2 above)
Comment (himorin) [reply index.html]: text updated per discussion within JL-TF. 'untransparent' has gone.

**Review by kidayasuo — APPROVED** [review]
Changes look good. thank you!

**Comment by kidayasuo** [comment]
Is the only change adding the PDF after my approval?

**Comment by himorin** [comment]
> Is the only change adding the PDF after my approval?

Update of PDF is not limited to this PR, since it's not updated as diff but by one binary, so PDF will be updated in parallel as https://github.com/w3c/simple-ruby/pull/83

**Review by kidayasuo — APPROVED** [review]
I am fine with the fix. thank you!

File: index.html [hunk 3]
```diff
@@ -209,18 +209,24 @@ <h3 id="matters-considered-by-the-placement-rules">Considerations for  simple pl
...
-    Also, when the ruby block is placed at 
-    the line head or the line end the method used in this document does not align the first or last 
-    character of the base text to the line head or the line 
-    end by modifying the relative positions of the ruby annotation and its base text. 
-    To summarise, the relative positions decided in the first step are not 
-    modified by the second step at all.</li>
+<li id="l20200529009">Processing is done in two levels.
+    The first level only determines relative positions of ruby annotation and its base.
+    Meanwhile, the second level determines inline positions of <a>ruby boxes</a>.
+    In other words, relative positions of the ruby annotation and its base text as decided in the first level
```
Comment (r12a) [inline index.html:215]: This is a very, very long sentence, and gets a little confusing towards the end.  Can it be broken up?  For example, we might be able to say: 
In other words, the relative positions of the ruby annotation and its base text, as decided in the first level, are not referenced nor modified by the second level. For example, this is not affected by..., except where....

File: index.html [hunk 4]
```diff
@@ -209,18 +209,24 @@ <h3 id="matters-considered-by-the-placement-rules">Considerations for  simple pl
...
-    following characters. In other words relative positions of the ruby annotation and its base text as decided in the first step are not modified 
-    in  light of any characters preceding and following the ruby block. 
-    Also, when the ruby block is placed at 
-    the line head or the line end the method used in this document does not align the first or last 
-    character of the base text to the line head or the line 
-    end by modifying the relative positions of the ruby annotation and its base text. 
-    To summarise, the relative positions decided in the first step are not 
-    modified by the second step at all.</li>
+<li id="l20200529009">Processing is done in two levels.
+    The first level only determines relative positions of ruby annotation and its base.
```
Comment (r12a) [inline index.html:213]: suggestion:
...relative positions of each ruby annotation...

File: index.html (hunk 3 above)
Comment (r12a) [inline index.html:215]: I think we can drop 'Meanwhile'.  Either that or replace it with 'On the other hand' if you want to stress the distinction, though i think having 2 levels already does that.

Comment (r12a) [reply index.html:215]: I just read the previous version of this sentence, and it seems much clearer.  What was the goal in rewriting it?

File: index.html [hunk 5]
```diff
@@ -209,18 +209,24 @@ <h3 id="matters-considered-by-the-placement-rules">Considerations for  simple pl
...
+    Meanwhile, the second level determines inline positions of <a>ruby boxes</a>.
+    In other words, relative positions of the ruby annotation and its base text as decided in the first level
+    are not referenced nor modified by the second level such as in light of any characters or line edge preceding and following the <a>ruby box</a>, 
+    except for hanging length of the ruby annotation from its base text when the ruby annotation is wider than its base text 
+    which may be used by the second level for removing spacing of the characters preceding or following the <a>ruby box</a>.
+    Two levels are processed as following:
+    <ol>
+    <li id="l20220318001">The first level takes pairs of ruby base and its annotation
```
Comment (r12a) [inline index.html:221]: The numbered list items here seem to just repeat what was already said.  Are they useful?

Comment (r12a) [reply index.html:221]: The phrase " and outputs styled text" seems a little vague.  What is styled, and how? Is the meaning simply that it outputs a ruby block that is integrated with the surrounding text?

**Review by r12a — CHANGES_REQUESTED** [review]
In addition to the specific comments here, i think the document needs a thorough edit for English wording. I may be able to help, if i can find time for it, but probably best wait until these and other issues are resolved.

We should also not forget to ask the i18n WG to review the document at some point - probably after the edit for English, to avoid getting many reports of editorial nits.

File: index.html (hunk 4 above)
Comment (himorin) [reply index.html:213]:
```suggestion
    The first level only determines relative positions of each ruby annotation and its base.
```

File: index.html [hunk 6]
```diff
@@ -209,18 +209,24 @@ <h3 id="matters-considered-by-the-placement-rules">Considerations for  simple pl
...
-    in  light of any characters preceding and following the ruby block. 
-    Also, when the ruby block is placed at 
-    the line head or the line end the method used in this document does not align the first or last 
-    character of the base text to the line head or the line 
-    end by modifying the relative positions of the ruby annotation and its base text. 
-    To summarise, the relative positions decided in the first step are not 
-    modified by the second step at all.</li>
+<li id="l20200529009">Processing is done in two levels.
+    The first level only determines relative positions of ruby annotation and its base.
+    Meanwhile, the second level determines inline positions of <a>ruby boxes</a>.
```
Comment (himorin) [inline index.html:214]:
```suggestion
    The second level determines inline positions of <a>ruby boxes</a>.
```

**Comment by kidayasuo** [comment]
If we already have a fix that is reasonable we can go ahead. Otherwise we can process the original issue as not-to-be-fixed. What do you think?
